cni/utils/errutils.go

```diff
@@ -0,0 +1,20 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/Azure/azure-container-networking/cns/cnsclient"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+)
+
+// TODO : Move to common directory like common, after fixing circular dependencies
+func IsNotFoundError(err error) bool {
+	switch err := err.(type) {
+	case *cnsclient.CNSClientError:
+		var cnsError *cnsclient.CNSClientError
+		// not expected to fail
+		_ = errors.As(err, &cnsError)
+		return (cnsError.Code == restserver.UnknownContainerID)
+	}
```

Comment:
When you do a type switch and assign to a variable, you can use that variable as the concrete type directly within each case block. If you don't want to shadow `err` to make this more evident, you can do something like
```go
switch e := err.(type) {
case *cnsclient.CNSClientError:
    return e.Code == restserver.UnknownContainerID
}
```